src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/SuspiciousCastFromCharToIntTests.cs

```diff
@@ -0,0 +1,225 @@
+    }}
+}}";
+
+            return VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Theory]
+        [InlineData("c, n, o")]
+        [InlineData("c, (int)c, o")]
+        [InlineData("s, n, o")]
+        [InlineData("s, (int)c, o")]
+        public Task StringSplit_NoImplicitConversion_NoDiagnostic_CS(string argumentList)
+        {
+            var test = new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        $@"
+using System;
+public class Testopolis
+{{
+    public void M(string s, char c, int n, StringSplitOptions o)
+    {{
+        nameof(M).Split({argumentList});
+    }}
+}}"
+                    },
+                    ReferenceAssemblies = ReferenceAssemblies.Net.Net50
+                }
+            };
+
+            return test.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("{|#0:c|}", "c.ToString()")]
+        [InlineData("{|#0:'a'|}", @"""a""")]
+        [InlineData("{|#0:capacity: c|}", "c.ToString()")]
+        [InlineData("{|#0:capacity: 'a'|}", @"""a""")]
+        public Task StringBuilderCtor_Int32_Fixed_CS(string testArgumentList, string fixedArgumentList)
+        {
+            string format = @"
+using System.Text;
+public class Testopolis
+{{
+    public void M(string s, char c, int n)
+    {{
+        new StringBuilder({0});
+    }}
+}}";
+            var test = new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources = { string.Format(CultureInfo.InvariantCulture, format, testArgumentList) },
+                    ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                    ExpectedDiagnostics = { VerifyCS.Diagnostic(Rule).WithLocation(0) }
+                },
+                FixedState =
+                {
+                    Sources = { string.Format(CultureInfo.InvariantCulture, format, fixedArgumentList) },
+                    ReferenceAssemblies = ReferenceAssemblies.Net.Net50
+                }
+            };
+
+            return test.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("{|#0:c|}, n")]
+        [InlineData("n, {|#0:c|}")]
+        [InlineData("{|#0:c|}, {|#1:c|}", 2)]
+        [InlineData("s, {|#0:c|}")]
+        [InlineData("s, n, n, {|#0:c|}")]
+        [InlineData("s, {|#0:c|}, n, n")]
+        [InlineData("s, {|#0:c|}, {|#1:c|}, {|#2:c|}", 3)]
+        public Task StringBuilderCtor_NonFixable_Diagnostic_CS(string argumentList, int diagnosticsCount = 1)
+        {
+            string source = $@"
+using System.Text;
+public class Testopolis
+{{
+    public void M(string s, char c, int n)
+    {{
+        new StringBuilder({argumentList});
+    }}
+}}";
+            var diagnostics = Enumerable.Range(0, diagnosticsCount).Select(x => VerifyCS.Diagnostic(Rule).WithLocation(x)).ToArray();
+
+            return VerifyCS.VerifyAnalyzerAsync(source, diagnostics);
```

Comment:
Use VerifyCodeFixAsync even for non-fixable diagnostics. This ensures that we're not offering an "empty" codefix.

Reply:
Fixed.